src/windows/osd.c

```diff
@@ -455,15 +454,33 @@ int getifaddrs(struct ifaddrs **ifap)
 			if (aa->IfType == IF_TYPE_SOFTWARE_LOOPBACK)
 				fa->ifa_flags |= IFF_LOOPBACK;
 
-			fa->ifa_addr = (struct sockaddr *) &fa->in_addr;
-			fa->ifa_netmask = (struct sockaddr *) &fa->in_netmask;
+			fa->ifa_addr = (struct sockaddr *) &fa->in_addrs;
+			fa->ifa_netmask = (struct sockaddr *) &fa->in_netmasks;
 			fa->ifa_name = fa->ad_name;
 
-			fa->in_netmask.sin_family = pSockAddr->sa_family;
-			fa->in_addr.sin_family = pSockAddr->sa_family;
-			fa->in_netmask.sin_addr.S_un.S_addr = mask;
-			pInAddr = (struct sockaddr_in *) pSockAddr;
-			fa->in_addr.sin_addr = pInAddr->sin_addr;
+			if (pSockAddr->sa_family == AF_INET && check_ipv6 == 0) {
+				subnet = 0;
+				mask = &subnet;
+				if (ConvertLengthToIpv4Mask(ua->OnLinkPrefixLength, mask) !=
+					NO_ERROR) {
+					ret = -FI_ENODATA;
+					goto out;
+				}
+				struct sockaddr_in *addr4 = (struct sockaddr_in *)
+							    &fa->in_addrs;
+				struct sockaddr_in *netmask4 = (struct sockaddr_in *)
+								&fa->in_netmasks;
+				netmask4->sin_family = pSockAddr->sa_family;
+				addr4->sin_family = pSockAddr->sa_family;
+				netmask4->sin_addr.S_un.S_addr = mask;
+				pInAddr = (struct sockaddr_in *) pSockAddr;
+				addr4->sin_addr = pInAddr->sin_addr;
+			}
+			if (pSockAddr->sa_family == AF_INET6 && check_ipv6 == 1) {
```

Comment:
This should be an else or else-if.  If this is an else-if, then there should be an else to handle the alternate case.

Reply:
@shefty Done. AF_INET6 is for else branch.